packages/schemas/alterations/next-1668140465-support-multiple-connector-instances.ts

```diff
@@ -0,0 +1,26 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      DELETE FROM connectors WHERE config <@ '{}'::jsonb;
+      ALTER TABLE connectors ADD COLUMN sync_profile boolean NOT NULL DEFAULT false;
+      ALTER TABLE connectors ADD COLUMN connector_id varchar(128);
+      UPDATE connectors SET connector_id = id;
+      ALTER TABLE connectors ALTER COLUMN connector_id SET NOT NULL;
+      ALTER TABLE connectors ADD COLUMN metadata jsonb NOT NULL DEFAULT '{}'::jsonb;
+    `);
+  },
+  down: async (pool) => {
+    await pool.query(sql`
+      DELETE FROM connectors WHERE id <> connector_id;
```

Comment:
Will this case happen?

Reply:
We can have multiple rows with the same `connector_id` (multiple instances). All connectors with specific `connector_id` added after migrating to the new design are with randomly generated `id` to make them unique. In this case, those rows with `id` equal to `connector_id` have existed in the old design.